src/openloco/windows/music_selection.cpp

```diff
@@ -0,0 +1,204 @@
+        window->colours[0] = interface->colour_0B;
+        window->colours[1] = interface->colour_10;
+
+        window->row_count = audio::num_music_tracks;
+        window->row_hover = -1;
+
+        return window;
+    }
+
+    // 0x004C165D
+    static void draw(ui::window* window, gfx::drawpixelinfo_t* dpi)
+    {
+        // Draw widgets.
+        window->draw(dpi);
+    }
+
+    // 0x004C1663
+    static void draw_scroll(ui::window* window, gfx::drawpixelinfo_t* dpi, uint32_t scrollIndex)
+    {
+        auto shade = colour::get_shade(window->colours[1], 4);
+        gfx::clear_single(*dpi, shade);
+
+        auto config = config::get();
+
+        uint16_t y = 0;
+        for (uint16_t i = 0; i < window->row_count; i++)
+        {
+            string_id text_colour_id = string_ids::white_stringid2;
+
+            // Draw hovered track
+            if (i == window->row_hover)
+            {
+                gfx::draw_rect(dpi, 0, y, 800, rowHeight, 0x2000030);
+                text_colour_id = string_ids::wcolour2_stringid2;
+            }
+
+            // Draw checkbox.
+            gfx::fill_rect_inset(dpi, 2, y, 11, y + 10, window->colours[1], 0xE0);
+
+            // Draw checkmark if track is enabled.
+            if (config.enabled_music[i])
+                gfx::draw_string_494B3F(*dpi, 2, y, window->colours[1], string_ids::wcolour2_stringid2, (void*)&string_ids::checkmark);
+
+            // Draw track name.
+            string_id music_title_id = audio::getMusicInfo(i)->title_id;
+            gfx::draw_string_494B3F(*dpi, 15, y, window->colours[1], text_colour_id, (void*)&music_title_id);
+
+            y += rowHeight;
+        }
+    }
+
+    // 0x004C176C
+    static void get_scroll_size(ui::window* window, uint32_t scrollIndex, uint16_t* scrollWidth, uint16_t* scrollHeight)
+    {
+        *scrollHeight = rowHeight * audio::num_music_tracks;
+    }
+
+    // 0x004C1757
+    static void on_mouse_up(ui::window* window, widget_index widgetIndex)
+    {
+        switch (widgetIndex)
+        {
+            case widx::close:
+                WindowManager::close(window->type);
+                break;
+        }
+    }
+
+    // 0x004C1799
+    static void on_scroll_mouse_down(ui::window* window, int16_t x, int16_t y, uint8_t scroll_index)
+    {
+        uint16_t currentTrack = y / rowHeight;
+        if (currentTrack > window->row_count)
+            return;
+
+        auto& config = config::get();
+
+        // Toggle the track in question.
+        config.enabled_music[currentTrack] = !config.enabled_music[currentTrack];
+
+        // Are any tracks enabled?
+        uint8_t anyEnabled = 0;
+        for (uint8_t i = 0; i < audio::num_music_tracks; i++)
+            anyEnabled |= config.enabled_music[i];
+
+        // Ensure at least this track is enabled.
+        if (!anyEnabled)
+            config.enabled_music[currentTrack] = true;
+
+        config::write();
+        call(0x048AA0C);
```

Comment:
This function is being called by both the options screen and this window. Could you implement this as well? That should tackle all functions accessing the MusicInfo struct